api/v1beta1/disruption_cron_webhook.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package v1beta1
+
+import (
+	"errors"
+
+	"github.com/DataDog/chaos-controller/utils"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var (
+	disruptionCronWebhookLogger     *zap.SugaredLogger
+	disruptionCronWebhookRecorder   record.EventRecorder
+	disruptionCronWebhookDeleteOnly bool
+)
+
+func (d *DisruptionCron) SetupWebhookWithManager(setupWebhookConfig utils.SetupWebhookWithManagerConfig) error {
+	disruptionCronWebhookRecorder = setupWebhookConfig.Recorder
+	disruptionCronWebhookDeleteOnly = setupWebhookConfig.DeleteOnlyFlag
+	disruptionCronWebhookLogger = setupWebhookConfig.Logger.With("source", "admission-controller")
+
+	return ctrl.NewWebhookManagedBy(setupWebhookConfig.Manager).
+		For(d).
+		Complete()
+}
+
+//+kubebuilder:webhook:webhookVersions={v1},path=/validate-chaos-datadoghq-com-v1beta1-disruptioncron,mutating=false,failurePolicy=fail,sideEffects=None,groups=chaos.datadoghq.com,resources=disruptioncrons,verbs=create;update;delete,versions=v1beta1,name=vdisruptioncron.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &DisruptionCron{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (d *DisruptionCron) ValidateCreate() (admission.Warnings, error) {
+	log := disruptionCronWebhookLogger.With("disruptionCronName", d.Name, "disruptionCronNamespace", d.Namespace)
+
+	log.Debugw("validating created disruption cron", "spec", d.Spec)
+
+	// delete-only mode, reject everything trying to be created
+	if disruptionCronWebhookDeleteOnly {
+		return nil, errors.New("the controller is currently in delete-only mode, you can't create new disruption cron for now")
+	}
+
+	// send informative event to disruption cron to broadcast
+	disruptionCronWebhookRecorder.Event(d, Events[EventDisruptionCronCreated].Type, string(EventDisruptionCronCreated), Events[EventDisruptionCronCreated].OnDisruptionTemplateMessage)
+
+	return nil, nil
+}
+
+func (d *DisruptionCron) ValidateUpdate(_ runtime.Object) (warnings admission.Warnings, err error) {
+	log := logger.With("disruptionCronName", d.Name, "disruptionCronNamespace", d.Namespace)
+
+	log.Debugw("validating updated disruption cron", "spec", d.Spec)
+
+	// delete-only mode, reject everything trying to be created
```

Comment:
```suggestion
	// delete-only mode, reject everything trying to be updated
```
Same comment as above here.